torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -370,12 +358,24 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
                 from a call to :meth:`state_dict`
         """
 
-        # Check whether we got a local or global dict
-        if "local_state_dict" in state_dict and state_dict["local_state_dict"]:
-            self.load_local_state_dict(state_dict)
-        else:
-            # Dispatch this rank's state dictionary to the wrapped shard optimizer
-            self.load_local_state_dict(ZeroRedundancyOptimizer.rank_local_state_dict(self.rank, state_dict))
+        # NOTE: PyTorch 1.5 does not index linearly but with the id(params) at saving time
+        # we work around that here by using the fact that the params are ordered as in the param_groups
+        pytorch15_index_redirect = {k: i for i, k in enumerate(state_dict["state"].keys())}
```

Comment:
do we need this v1.5 work around here? I assume since this optimizer is added in 1.8, this feature does not exist in v1.5. I think it makes sense to have such work around in FAIRScale as it needs to be compatible with multiple PyTorch versions, but this is already in PyTorch, so maybe we can skip this?

Reply:
Or are we assuming there can be use cases where users need to load a v1.5 checkpoint into v1.8?

Reply:
:) yeah that was for the second point, but maybe overkill ? I'm completely ok with removing that, no problem

Reply:
I'd vote for removing it for simplicity. If there are indeed users hitting this problem, we can add this back later. 

Would I be correct if I assume that a more common use case would be loading model checkpoints saved from a different PT version instead of optimizer state checkpoints? And it would still work without this PT v1.5 redirect if users load model checkpoints from v1.5 and then create the ZeRO optimizer instance, as the optimizer no longer touches the checkpoint?